examples/abitest/tests/src/tests.rs

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright 2019 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use abitest_frontend::proto::abitest::{ABITestRequest, ABITestResponse};
+use oak::grpc;
+use oak::OakStatus;
+use oak_tests::proto::manager::{
+    ApplicationConfiguration, Channel, Channel_Endpoint, GrpcServerNode, Node,
+    Node_oneof_node_type, WebAssemblyNode,
+};
+use std::collections::HashMap;
+
+fn test_config() -> ApplicationConfiguration {
+    let mut grpc = Node::new();
+    grpc.set_node_name("grpc_server".to_string());
+    grpc.node_type = Some(Node_oneof_node_type::grpc_server_node(GrpcServerNode::new()));
+    let mut frontend = Node::new();
+    frontend.set_node_name("frontend".to_string());
+    let mut wasm_node = WebAssemblyNode::new();
+    wasm_node.set_wasm_contents_name("frontend-code".to_string());
+    frontend.node_type = Some(Node_oneof_node_type::web_assembly_node(wasm_node));
+    let mut grpc_channel = Channel::new();
+    {
+        let mut src_endpoint = Channel_Endpoint::new();
+        src_endpoint.set_node_name("grpc_server".to_string());
+        src_endpoint.set_port_name("req".to_string());
+        grpc_channel.set_source_endpoint(src_endpoint);
+        let mut dest_endpoint = Channel_Endpoint::new();
+        dest_endpoint.set_node_name("frontend".to_string());
+        dest_endpoint.set_port_name("gRPC_input".to_string());
```

Comment:
IIRC `gRPC_input` was a "well known" port name. Could you create a const somewhere and reference it directly?

Reply:
This one is deliberately *not* the default port name; added a comment to that effect (and a constant)